packages/hemera/lib/index.js

```diff
@@ -202,6 +204,65 @@ class Hemera {
     }
   }
 
+  /**
+   *
+   *
+   * @memberof Hemera
+   */
+  _registerEvents() {
+    this._transport.driver.on('error', err => {
+      this.log.error(err, 'Could not connect to NATS!')
+      this.log.error("NATS Code: '%s', Message: %s", err.code, err.message)
+
+      // Exit only on connection issues.
+      // Authorization and protocol issues don't lead to process termination
+      if (natsConnCodes.indexOf(err.code) > -1) {
+        // We have no NATS connection and can only gracefully shutdown hemera
+        this.close()
+      }
+      
+      this._isReady=false;
+    })
+
+    this._transport.driver.on('permission_error', err => {
+      this.log.error(err, 'NATS permission error')
+      this._isReady=false;
+    })
+
+    this._transport.driver.on('reconnect', () => {
+      this.log.info('NATS reconnected!')
+      this._isReady=false;
+    })
+
+    this._transport.driver.on('reconnecting', () => {
+      this.log.warn('NATS reconnecting ...')
+      this._isReady=false;
+    })
+
+    this._transport.driver.on('disconnect', () => {
+      this.log.warn('NATS disconnected!')
+      this._isReady=false;
+    })
+
+    // when nats was not able to reconnect or connection was closed due to other reasons
+    // the process should die and restarted
+    this._transport.driver.on('close', () => {
+      this.log.error(new Errors.HemeraError('NATS connection closed!'))
+      this._isReady=false;
+    })
+
+    this._transport.driver.on('connect', () => {
+      this.log.info('Connected!')
+
+      this._isReadyWaits.forEach((readyCall)=>{
```

Comment:
I think this doesn't work because you call `_registerEvents` in the constructor. At that time no user has registered a ready callback handler.